### PR TITLE
#170947685 allow suppliers to add their accommodation

### PR DESCRIPTION
**Pull request by Kagorora**
#### What does this PR do?

- Have the suppliers add their accommodation on the app

#### Description of Task to be completed?

-  logging in the system as `suppliers` 

-  validate if they are `suppliers` or `admin`

-  create accommodation

#### How should this be manually tested?

- **Login as a supplier,** 

    POST        `http://localhost:4000/api/v1/auth/login`

   { 
	"email": "blackhotel@gmail.com", 
	"password": "Niyonkuru@1" 
   }

- **Create accomodation**

    POST        `http://localhost:4000/api/v1/create/accommodation`

    {
 "accommodationName":"Black Hotel",
 "locationName": "Kigali",
 "streetNumber": "kg102 ",
   "numberOfRooms": 100,
   "description":"it has wifi and big secreen TV",
    "availableRooms": [
      {
        "roomName": "double room",
        "roomType": "king",
        "price":"$50/night"
      },
       {
        "roomName": "sleeping room",
        "roomType": "qween",
        "price":"$53/night"
      },
       {
        "roomName": "sleeping room",
        "roomType": "qween",
        "price":"$54/night"
      }
      ,
       {
        "roomName": "sleeping room",
        "roomType": "qween",
        "price":"$55/night"
      }
    ]
  }

- ** Update the accommodation**

     PATCH     `http://localhost:4000/api/v1/edit/accommodation/2`

   {
            "reason": "learn how I can get a job at google",
            "origin": "Kigali"
   }

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?

- [170947684](https://www.pivotaltracker.com/story/show/170947684)

#### Screenshots (if appropriate)

1. **Login as a supplier**

Screen Shot 2020-03-10 at 16 31 48

2.  **create an accommodation**

Screen Shot 2020-03-10 at 16 33 03

3. **Trying to access the route while you not either a manager not an admin**

  - ** Login as a requester or manager

Screen Shot 2020-03-10 at 16 43 12

 - **Try to create an accommodation**
Screen Shot 2020-03-10 at 16 35 16

4. **Suppliers can update their accommodation**

Screen Shot 2020-03-10 at 16 51 54

5. ** suppliers can upload picture of their accommodation**

Screen Shot 2020-03-10 at 17 09 18

#### Questions:

- N/A

### CODE REVIEW

File: src/tests/accommodation/createAccommodationFacility.test.js [hunk 1]
```diff
@@ -194,6 +210,27 @@ export const editAccommodations = () => {
...
+    });
+
+    it('it should return 200 when editing single accommodation ', (done) => {
+      chai
+        .request(app)
+        .patch(`/api/v1/edit/accommodation/${2}`)
+        .send(editAccommodation)
+        .end((err, res) => {
+          expect(res.statusCode).to.equal(200);
+          expect(res.body.data).to.be.an('object');
```
Comment (T2Wil) [inline src/tests/accommodation/createAccommodationFacility.test.js:229]: @Kagorora , you did a wonderful job here.
As a suggestion, I think you might consider expecting some of the object's variables/values to make the test more firm.

Comment (Kagorora) [reply src/tests/accommodation/createAccommodationFacility.test.js:229]: @T2Wil  as you can see below this comment on line 230 and 231 I have implemented your feedback

**Review by T2Wil — APPROVED** [review]

**Review by Niyonsengaeric — APPROVED** [review]

File: src/controllers/accommodation.js [hunk 2]
```diff
@@ -2,11 +2,10 @@ import { config } from 'dotenv';
 import models from '../db/models';
 
 config();
-
 export default class accomodationFacility {
   static async getAllAccommodations(req, res) {
     try {
-      const accommodations = await models.Accommodation.findAll({ where: { userId: req.user.id } });
+      const accommodations = await models.Accommodation.findAll();
```
Comment (kagabof) [inline src/controllers/accommodation.js:8]: can you explain the reason you changed the findAll conditions?

**Review by kagabof — CHANGES_REQUESTED** [review]

File: src/controllers/accommodation.js (hunk 2 above)
Comment (Kagorora) [reply src/controllers/accommodation.js:8]: @kagabof  the reason is, all the users should be able to to view available accommodation. before it was only seen by the person who have created them either the manager or the supplier but considering that Users will need to view those accommodation to rate them or any other reason so I felt that it should be seen by every authenticated user

Comment (kagabof) [reply src/controllers/accommodation.js:8]: Okay.

**Review by kagabof — APPROVED** [review]

**Pull request merged**